brave/app/scripts/controllers/preferences.js

```diff
@@ -0,0 +1,737 @@
+const ObservableStore = require('obs-store')
+const normalizeAddress = require('eth-sig-util').normalize
+const { isValidAddress, sha3, bufferToHex } = require('ethereumjs-util')
+const extend = require('xtend')
+
+
+class PreferencesController {
```

Comment:
Looks like there are some copied files here. that would make it even more difficult for rebasing. 
We'll want things like subclasses here that take the input code and modifies it in the needed ways instead.